gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt

```diff
@@ -50,7 +51,21 @@ class ComposeCompilerKotlinSupportPlugin @Inject constructor(
         val service = ComposeMultiplatformBuildService.provider(target)
         buildEventsListenerRegistry.onTaskCompletion(service)
 
-        service.get().parameters.unsupportedCompilerPlugins.add(
+        val providedService = try {
+            service.get()
+        } catch (e: ClassCastException) {
+            // Compose Gradle plugin was probably loaded more than once
+            // See https://github.com/JetBrains/compose-multiplatform/issues/3459
+
+            throw IllegalStateException(
+                "Failed to get ComposeMultiplatformBuildService instance." +
+                        " Compose Gradle plugin was probably loaded more than once." +
+                        " Consider declaring it in the root build.gradle.kts. Original exception message: ${e.message}",
```

Comment:
I'm not sure "Original exception message" is necessary.
`--stacktrace` or `--full-stacktrace` will print the nested exception.